### DISCUSSION TITLE
Missing so-dockerregistry after attempted upgrade to 2.3.70

### DISCUSSION THREAD

**Discussion by pimpernoni** [Security-Onion-Solutions/securityonion#5655] [Unsupported Versions]
I have been stumped on how to resolve an issue that arose during an attempted upgrade from 2.3.61 to 2.3.70. I believe the problem arose out of my Suricata IDS on a SOHO pfSense router blocking some of the IPs for the upgrade, and consequently interrupted the upgrade. Now I get the error:
"Updating dockers to 2.3.70.
Error response from daemon: No such container: so-dockerregistry
Soup failed with error 1: Unhandled error
Unhandled error occured, please check /root/soup.log for details."
The soup.log doesn't contain much detail, and I can't see a clean way to download the so-dockerregistry from the repository. 

so-status shows:
Docker---OK
soaptcacherng---OK
but 
so-curator---WAIT_START and
same for the rest below.

I get a compile error on salt-call state.highstate.

Relevant Logs are attached

This is a Standalone install downloaded from SO-iso CentOS onto PC box with Intel i5 3200 Hz, 32 G RAM, 1 TB RAID1 /nsm /etc/soversion = 2.3.61

Any way to resurrect with SSH docker surgery? I have a lot of tuned Suricata rules and 100+ days of logs that I would hate to have to scrap on a fresh reload. I will go that route if you say this is the best, but just crossing fingers.

Thanks for any guidance
[soup.log.txt](https://github.com/Security-Onion-Solutions/securityonion/files/7230739/soup.log.txt)
[salt.master.txt](https://github.com/Security-Onion-Solutions/securityonion/files/7230740/salt.master.txt)

**Comment by dougburks** [comment]
Try `sudo so-docker-refresh`.

**Comment by m0duspwnens** [comment]
Hi @pimpernoni. It looks like there is possibly a duplicate sid defined in the thresholding pillar which is causing Salt states to fail and soup as well. If you resolve the issue with the pillar for the minion mentioned in the log, then soup should run properly. 

I am going to open an issue referencing this discussion as soup should handle the error state more appropriately.

**Reply by pimpernoni** [reply]
Thanks for the suggestion, and I did try the so-docker-refesh but I seem to be encountering a problem on the local machine that does not resolve with this command. I have packet-captured during the exec phase, and I seem to be able to download content into the box but it is not passing into the Docker images. I have attached the output from the so-docker-refresh, so I am at a loss as to what to tweek next.
Thanks for pointing me in the right direction.

P
[so-docker-refresh.txt](https://github.com/Security-Onion-Solutions/securityonion/files/7349857/so-docker-refresh.txt)

**Reply by Git-Me-Some-Hub** [reply]
This is not super helpful, but I had a very similar thing happen to me when an update was blocked because a Nutanix VMagent was using the dockerregistry port number (port:5000). I actually had to rebuild the cluster to get it back working. 
https://github.com/Security-Onion-Solutions/securityonion/discussions/4901

**Reply by Mooshcakes** [reply]
Was this issue ever resolved? 

Ran into the exact same issues on our SO2 stack. Failed soup led to so-dockerregistry [missing] and so-docker-refresh results in the same error. Have tried everything I can find in the discussion boards. Is the only solution to rebuild?

Upgrading from 2.3.91 to 2.3.100

**Reply by dougburks** [reply]
@Mooshcakes From https://github.com/Security-Onion-Solutions/securityonion/discussions/1720:
> Start a new discussion instead of replying to somebody else's discussion
Please search to see if you can find similar discussions that may help you. However, in order to avoid confusion, please do NOT reply to somebody else's discussion with your own issue. Instead, please start a new discussion and in that new discussion you can provide a hyperlink to the related discussion.

**Reply by Git-Me-Some-Hub** [reply]
> Was this issue ever resolved?
> 
> Ran into the exact same issues on our SO2 stack. Failed soup led to so-dockerregistry [missing] and so-docker-refresh results in the same error. Have tried everything I can find in the discussion boards. Is the only solution to rebuild?
> 
> Upgrading from 2.3.91 to 2.3.100

I did not get this working, I ended up totally rebuilding my environment from a snapshot… that experience was also not amazing as I had some data loss for some odd reason. I do think there is another discussion out there that someone might have fixed theirs, though.

**Reply by Mooshcakes** [reply]
Thanks for the response
